### PR TITLE
Sum digit

### PR DESCRIPTION
**Pull request by Okazia1715**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/digit_sum.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on 06 01 25
+
+@author: Alona Niechvieieva
+"""
+
```
Comment (salihakalender) [inline solutions/digit_sum.py:8]: please add describe of purpose

#Module for calculating the sum of the digits of a natural number.

File: solutions/tests/test_digit_sum.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+@author: Alona Niechvieieva
+"""
+
+import unittest
+
+from ..digit_sum import digit_sum  # type: ignore # noqa: E402
+
+""" A module for testing the function that prints the sum of integers
+Module contents: TestTestDigitSum: unit tests for the `digit_sum` function.
```
Comment (salihakalender) [inline solutions/tests/test_digit_sum.py:14]: Can you please update  module docstring for the test file. The function computes, not prints, the sum of digits. 

#A module for testing the function that computes the sum of digits of a number.

File: solutions/digit_sum.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    Parameters:
+        number: int, greater than or equal to zero
+
+    Returns:
+        int: The sum of the digits of the number
+
+    Raises:
+        TypeError: If number is not an integer
+        ValueError: If number is less than 0
+
```
Comment (salihakalender) [inline solutions/digit_sum.py:23]: Your docstring explanations are excellent. However, in the 'Raises' section, you mention TypeError and ValueError, but the function actually raises AssertionError. Please update this:

 AssertionError: If an assumption about the input fails.

File: solutions/digit_sum.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on 06 01 25
+
+@author: Alona Niechvieieva
+"""
+
+
+def digit_sum(number):
```
Comment (salihakalender) [inline solutions/digit_sum.py:10]: The function does not include type annotations. Could you please add type hints for clarity.

def digit_sum(number: int) -> int:

File: solutions/tests/test_digit_sum.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        # It should raise an assertion error if the argument is less than 0
+        with self.assertRaises(AssertionError):
+            digit_sum(-1)
+
+    def test_not_an_integer(self):
+        # It should raise an assertion error if the argument is not an integer
+        with self.assertRaises(AssertionError):
+            digit_sum(1.0)
+
+
```
Comment (salihakalender) [inline solutions/tests/test_digit_sum.py:47]: You might add a brief comment here

#Run the unit tests when the script is executed directly

**Review by salihakalender — CHANGES_REQUESTED** [review]
Your function and unit tests are well-structured and demonstrate a clear understanding of coding principles. There are a few small changes needed, and once those are addressed, your work will be even better. You’re doing an excellent job!

**Comment by Okazia1715** [comment]
> Your function and unit tests are well-structured and demonstrate a clear understanding of coding principles. There are a few small changes needed, and once those are addressed, your work will be even better. You’re doing an excellent job!

Thank you for your thoughtful comments. I have taken them into consideration and updated my function. Please review it, and if everything looks good, kindly approve it.

**Review by salihakalender — APPROVED** [review]
Excellent work!
